src/main/java/edu/kit/compiler/transform/IRExpressionVisitor.java

```diff
@@ -0,0 +1,222 @@
+        }
+        Node objectAddress = methodInvocationExpressionNode.getObject().get().accept(pointerVisitor);
+        // collect arguments in list with this pointer
+        Node[] arguments = new Node[methodInvocationExpressionNode.getArguments().size() + 1];
+        arguments[0] = objectAddress;
+        for (int i = 1; i < methodInvocationExpressionNode.getArguments().size(); i++) {
+            Node n = methodInvocationExpressionNode.getArguments().get(i).accept(this);
+            arguments[i] = n;
+        }
+        int method = methodInvocationExpressionNode.getName();
+        Node methodAddress = getConstruction().newAddress(IRVisitor.getMethodContexts().get(method).getMethodEntity());
+
+        Node call = getConstruction().newCall(mem, methodAddress, arguments, m.getType());
+        Node tResult = getConstruction().newProj(call, Mode.getT(), Call.pnTResult);
+        Node res = getConstruction().newProj(tResult, m, 0);
+        Node projMem = getConstruction().newProj(call, Mode.getM(), Call.pnM);
+        getConstruction().setCurrentMem(projMem);
+
+        return res;
+    }
+
+    @Override
+    public Node visit(FieldAccessExpressionNode fieldAccessExpressionNode) {
+        Node fieldAddress = fieldAccessExpressionNode.accept(pointerVisitor);
+        Mode resultType = context.getTypeMapper().getDataType(fieldAccessExpressionNode.getResultType()).getMode();
+        return loadFromAddress(fieldAddress, resultType);
+    }
+
+    @Override
+    public Node visit(ArrayAccessExpressionNode arrayAccessExpressionNode) {
+        Node arrayAddress = arrayAccessExpressionNode.accept(pointerVisitor);
+        Mode resultType = context.getTypeMapper().getDataType(arrayAccessExpressionNode.getResultType()).getMode();
+        return loadFromAddress(arrayAddress, resultType);
+
+    }
+
+    private Node loadFromAddress(Node address, Mode type) {
+        Node mem = getConstruction().getCurrentMem();
+        Node load = getConstruction().newLoad(mem, address, type);
+
+        Node resMem = getConstruction().newProj(load, Mode.getM(), Load.pnM);
+        Node res = getConstruction().newProj(load, type, Load.pnRes);
+        getConstruction().setCurrentMem(resMem);
+        return res;
+    }
+
+    @Override
+    public Node visit(IdentifierExpressionNode identifierExpressionNode) {
+        Construction con = context.getConstruction();
+        if (identifierExpressionNode.getDefinition().getKind() == DefinitionKind.LocalVariable) {
+            Mode mode = getMode(identifierExpressionNode.getResultType());
+            return con.getVariable(context.getVariableIndex(identifierExpressionNode.getIdentifier()), mode);
+        } else if (identifierExpressionNode.getDefinition().getKind() == DefinitionKind.Parameter) {
+            return context.createParamNode(identifierExpressionNode.getIdentifier());
+        } else {
+            throw new IllegalArgumentException();
```

Comment:
You need to handle members, too (similar to `FieldAccess`, by calling the `PointerVisitor`)